README.md

```diff
@@ -1,12 +1,17 @@
 # hydra core API documentation
 
-The hydra header exposes interfaces that an implementation class can inherit to create an emulator.
+The hydra header exposes interfaces that an implementation class can inherit from to create an emulator. For example, an emulator that uses or can use OpenGL for rendering should inherit the `GlEmulatorInterface` interface which provides functions for hooking the emulator's GL renderer with the frontend.
 You can inherit as many or as few of these interfaces as you want, except for `IBase` which must always be inherited.
 
+Currently available interfaces:
+- BaseEmulatorInterface: Provides functions for most basic functionality such as loading a file. Cores should typically implement this.
+- GlEmulatorInterface: Provides functions for communicating between an emulator's OpenGL renderer and the frontend. Cores that can use OpenGL for rendering should implement this.
+- FrontendDrivenEmulatorInterface: Provides functions for emulators that expect to be driven by the frontend. Cores that work like this should inherit from it
```

Comment:
Interface names are now IBase, IOpenGlRendered and IFrontendDriven. Also cores *must* implement IBase, that's the base class that can be transformed into the other interfaces.